SPARCore/GADArticle.m

```diff
@@ -1,115 +1,115 @@
+static NSString *const API_KEY_CONTENT = @"content";
+static NSString *const API_KEY_DATE_PUBLISHED = @"datePublished";
+static NSString *const API_KEY_HEADER_IMAGE = @"headerImage";
+static NSString *const API_KEY_ARTICLE_ID = @"id";
+static NSString *const API_KEY_PUBLICATION_ID = @"publication";
+static NSString *const API_KEY_TITLE = @"title";
 
-static NSString *const API_HOSTNAME = @"https://g2j7qs2xs7.execute-api.us-west-2.amazonaws.com/";
-static NSString *const API_PREFIX = @"devstable";
-static NSString *const API_PUBLICATION_PATH = @"publications";
-static NSString *const API_ARTICLE_PATH = @"articles";
+static NSString *const API_QUERY_PAGE_TOKEN = @"pageToken";
 
-+ (void) articlesFromPublication: (NSString *)publicationId
-                                        completionHandler:(void(^_Nonnull)(NSArray<GADArticle *>
-                                                                           *_Nullable articles,
-                                                            NSError *_Nullable error))completion {
-    NSURL *queryURL = [GADArticle urlForArticlesFromPublication:publicationId];
++ (void) articlesForPublicationId: (NSString *)publicationId
+                    nextPageToken: (NSString * _Nullable)nextPageToken
+                   withCompletion:(void(^_Nonnull)(NSArray<GADArticle *>
+                                                   *_Nullable articles,
+                                                   NSString *_Nullable token,
+                                                   NSError *_Nullable error))completion {
+    GADPublication *pub = [GADPublication new];
+    pub.publicationId=publicationId;
+    NSURL *queryURL = [pub urlForArticles];
     
-    [GADRemoteModel fetchModelsWithParams:queryURL
-                          queryParameters:@{}
-                         modelTransformer:^(NSData *jsonData) {
-                            return [GADArticle articlesFromJSON:jsonData];
-                         }
-                        completionHandler:completion];
-
-}
-
-
-- (void) fetchFulltextWithCompletion: (void(^_Nonnull)(GADRemoteModel * *_Nullable model,
-                                                NSError *_Nullable error))completion {
-    NSURL *queryURL = [GADArticle createURLWithArticle:self.articleId publication:self.publicationId];
+    NSMutableDictionary *params=[NSMutableDictionary new];
+    if (nextPageToken) {
+        [params setObject:nextPageToken forKey:API_QUERY_PAGE_TOKEN];
+    }
     
-    [GADRemoteModel fetchModelsWithParams:queryURL
-                          queryParameters:@{}
-                         modelTransformer:^(NSData *jsonData){
-                             return [GADArticle articlesFromJSON:jsonData];
+    [super fetchModelsWithParams:queryURL
+                          queryParameters:params
+                         modelTransformer:^(NSArray<NSDictionary *> *jsonArray) {
+                            return [self articlesFromArray:jsonArray];
```

Comment:
If we move this to `GADPublication`, I'll concede to allow `articlesFromArray:` to be a public method. Thoughts?